File: apps/backend-go/internal/roles.go
package internal

func CanEdit(role string) bool {
	return role == "owner" || role == "editor"
}

func CanShare(role string) bool {
	return role == "owner"
}

func ValidRole(role string) bool {
	return role == "owner" || role == "editor" || role == "viewer"
}

File: apps/backend-java/src/main/java/com/example/docs/Roles.java
package com.example.docs;

public final class Roles {
  private Roles() {}

  public static boolean canEdit(String role) {
    return "owner".equals(role) || "editor".equals(role);
  }

  public static boolean canShare(String role) {
    return "owner".equals(role);
  }

  public static boolean valid(String role) {
    return "owner".equals(role) || "editor".equals(role) || "viewer".equals(role);
  }
}
